tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,191 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define BUFFER_DIRECTORY FLB_TESTS_DATA_PATH "data/s3_local_buffer/"
+#define TEST_DATA "I love Fluent Bit"
+#define S3_KEY_FORMAT "logs/$TAG[2]/$TAG[0]/%Y/%m/%d"
+#define S3_OBJECT_KEY "logs/logs/ecs/2020/08/13"
+#define TAG "ecs.ec2.logs"
```

Comment:
Your test looks good- we will need a lot more test cases though

This test is also time dependent... it'll start failing tomorrow 😐

There's an easy fix to that though- hard code the time that you input to the function. You can take the time today (or the time of any date in recent history you like).

Reply:
Yes, I noticed this. Wanted to get some early feedback on the implementation.
I have already fixed this. I am also adding more tests for special character handling in s3 object keys.